overlay/default.nix

```diff
@@ -12,6 +12,48 @@ in
   # Uses sources.mix-to-nix and sources.gitignore
   nixUnstable = inputs.nix.defaultPackage.${final.system};
 
+  benchwrapper = prev.writeShellScriptBin "benchwrapper" ''
+    set -euo pipefail
+
+    # number of cores to run the wrapped program on
+    test_cpus=2
+
+    sctl=${prev.systemd}/bin/systemctl
+
+    #############
+
+    ncpus=$(( $(${prev.util-linux}/bin/lscpu -e | wc -l) - 1 ))
+
+    fullrange="0-$(( $ncpus - 1 ))"
+    sysrange="0-$(( $ncpus - $test_cpus - 1 ))"
+    shieldrange="$(( $ncpus - $test_cpus ))-$(( $ncpus - 1 ))"
+
+    cleanup() {
+      echo "called cleanup"
+      # restore full core utilization for all services
+      $sctl set-property --runtime system.slice AllowedCPUs="$fullrange"
+      $sctl set-property --runtime user.slice AllowedCPUs="$fullrange"
+    }
+
+    # make sure to allow scheduling on all cores again in case of error
+    trap cleanup EXIT
```

Comment:
yes, any kind of error code afaiu

Reply:
Hmm, I tried the following during my testing to load one of the CPUs:
```
benchwrapper sha1sum /dev/zero
```
A subsequent attempt to interrupt it with `Ctrl+C` exited the wrapper, but the created service was still running

Reply:
the services runs independent from the shell it was called in, only the outputs are piped.
it might be more intuitive to not daemonize it basically :thinking:

Reply:
Yes, but the question is whether this behaviour is expected by the user. IMO, it looks a bit confusing

Reply:
Perhaps, `cleanup` should do something like `systemctl kill run-u<id>.service`

Reply:
the latest change now attaches the process via a pseudo-tty to the calling shell so it can pass through the ctrl-c. just tried it with `benchwrapper 1 sha1sum /dev/zero` and it worked